Contents/Sketch/typography.js

```diff
@@ -8,12 +10,41 @@ var applyTypography = function (newStyles, sharedStyles) {
   };
 
   // this seems to be a simpler way to achieve the removeAllStyles function
-  [sharedStyles removeAllSharedObjects];
+  // [sharedStyles removeAllSharedObjects];
+
 
   for(var i=0; i<newStyles.length; i++) {
     createStyle(newStyles[i]);
   }
 
+
+  function checkForMatchingStyleAndMerge(existingStyleObjects, newStyleName, newStyle) {
+
+    for (var i=0; i<existingStyleObjects.count(); i++) {
+      // log(i);
+      var existingName = existingStyleObjects[i].name();
+
+      // log("existing name is " + existingName);
+      // log("new style name is " + newStyleName);
+
+      if(existingName == newStyleName) {
+        // log("they match");
+
+        sharedStyles.updateValueOfSharedObject_byCopyingInstance(existingStyleObjects[i], newStyle);
```

Comment:
Cool, I'll clean it up.

I used this: https://github.com/sketchplugins/plugin-requests/blob/master/Dump%20Selected%20Object.sketchplugin